src/legacy/core_plugins/data/public/query/query_bar/components/query_bar.test.tsx

```diff
@@ -102,4 +104,104 @@ describe('QueryBar', () => {
 
     expect(mockPersistedLogFactory.mock.calls[0][0]).toBe('typeahead:discover-kuery');
   });
+
+  it('Should render only timepicker when no options provided', () => {
+    const component = shallowWithIntl(
+      <QueryBar.WrappedComponent
+        onSubmit={noop}
+        appName={'discover'}
+        store={createMockStorage()}
+        intl={null as any}
+      />
+    );
+
+    expect(component.find(QUERY_INPUT_SELECTOR).length).toBe(0);
+    expect(component.find(TIMEPICKER_SELECTOR).length).toBe(1);
+  });
+
+  it('Should disable timepicker when asked', () => {
+    const component = shallowWithIntl(
+      <QueryBar.WrappedComponent
+        onSubmit={noop}
+        appName={'discover'}
+        store={createMockStorage()}
+        intl={null as any}
+        showDatePicker={false}
+      />
+    );
+
+    expect(component.find(QUERY_INPUT_SELECTOR).length).toBe(0);
+    expect(component.find(TIMEPICKER_SELECTOR).length).toBe(0);
+  });
+
+  it('Should render timepicker with options', () => {
```

Comment:
This doesn't seem to be testing anything different than the first test?

Reply:
It passes in the timepicker options explicitly.